lib/inventory/inventory_api.rb

```diff
@@ -246,8 +263,186 @@ def list_metrics_for_resource(resource, filter = {})
       with_nils.compact
     end
 
+    # Create a new feed
+    # @param [String] feed_id  Id of a feed - required
+    # @param [String] feed_name A display name for the feed
+    # @return [Object]
+    def create_feed(feed_id, feed_name = nil)
+      feed = create_blueprint
+      feed[:id] = feed_id
+      feed[:name] = feed_name
+
+      begin
+        return http_post('/feeds/', feed)
+      rescue => error
+        # 409 We already exist -> that is ok
+        if error.status_code == 409
+          the_feed = hawk_escape feed_id
+          return http_get('/feeds/' + the_feed)
+        else
+          raise
+        end
+      end
+    end
+
+    # Delete the feed with the passed feed id.
+    # @param feed Id of the feed to be deleted.
+    def delete_feed!(feed)
```

Comment:
IIRC, "dangerous method" is defined in Ruby as methods that change the instance itself, which is not the case here.